more_itertools/more.py

```diff
@@ -423,19 +423,10 @@ def unique_to_each(*iterables):
     It is assumed that the elements of each iterable are hashable.
 
     """
-    elements_to_indices = {}
     pool = [list(it) for it in iterables]
-    for i, it in enumerate(pool):
-        for element in it:
-            elements_to_indices.setdefault(element, set()).add(i)
-
-    for element, indices in iteritems(elements_to_indices):
-        if len(indices) != 1:
-            for i in indices:
-                while element in pool[i]:
-                    pool[i].remove(element)
-
-    return pool
+    counts = Counter(chain.from_iterable(map(set, pool)))
```

Comment:
We left `Counter` out originally because we were supporting Py2.6, but we're not at this point, so I think this is probably good.